### PR TITLE
Use conda-forge's xeus-cling

### PR DESCRIPTION
**Pull request by SylvainCorlay**

### CODE REVIEW

**Comment by certik** [comment]
@SylvainCorlay thanks for the PR!

@isuruf are the Travis errors caused by this PR? It doesn't seem so.

**Comment by SylvainCorlay** [comment]
I don't think the errors can be related to this PR.

I checked the binder on my branch and it appears to be working well.

**Comment by SylvainCorlay** [comment]
Merge?

**Comment by certik** [comment]
@SylvainCorlay I am worried to merge when tests fail. Let us first fix the failures, then merge this. I am hoping it's unrelated, so you don't have to worry about this.

I can see three different errors:

https://travis-ci.org/symengine/symengine/jobs/488915843

https://travis-ci.org/symengine/symengine/jobs/488915851

https://travis-ci.org/symengine/symengine/jobs/488915856

@isuruf any ideas?

**Comment by certik** [comment]
I am testing the latest master in #1505, so we'll see what happens.

**Comment by isuruf** [comment]
All of the errors are because of the compiler switch in conda-forge.

**Comment by certik** [comment]
I see. What is the solution? I don't want to have broken builds.

**Comment by SylvainCorlay** [comment]
Since you are using conda packages, you either have to use a recent enough gcc in your CI (>=7.3) or pin to the exact builds of the dependencies that were used before the switch.

**Comment by isuruf** [comment]
We can go back to the unmaintained `conda-forge/label/cf201901` channel which would bring back the old packages. Or fix the errors by tweaking the build scripts.

**Comment by isuruf** [comment]
All of the dependencies are C and shared libraries except for LLVM which is C++ and static. conda-forge switched ABI and uses a newer binutils and that's why only the LLVM builds fail.

**Comment by SylvainCorlay** [comment]
For the CI of xeus, which has a lot of binary dependencies, we moved to using conda-forge for everything including the compiler itself. But we are not testing that xeus can be built eith older ones anymore.

For xtensor, which is more complex but has less binary deps, we continue building with all supported compilers.

**Comment by certik** [comment]
@isuruf how about this plan:

1) Use conda-forge for everything including the compiler and all dependencies (LLVM, GMP, ...) on all platforms (Linux, Mac, Windows)

2) Use older or other compilers on the core of symengine only (no dependencies, besides GMP) to ensure our code can compile with older compilers

Libraries like LLVM would only be tested via 1).

Note: for 2) perhaps it make sense to ship some GMP equivalent ourselves that is short on number of lines to maintain (although not as performing), but that allows symengine to be tested without any dependencies, that way we can easily test older compilers on various platforms without worrying of compiling and handling dependencies. And in 2) we don't care about the dependencies anyway, as those are tested in 1).

**Comment by isuruf** [comment]
I disagree. Not testing older compilers with dependencies means there are codepaths not tested using older compilers and we may not notice issues with them.

**Comment by certik** [comment]
That is true, some dependency related code paths would not be tested with older compilers. However what is tested is the core without dependencies with older compilers, and all code paths (and dependencies) with newer compilers.

So the kinds of issues that could happen is that we use too modern C++ with some dependency. But it shouldn't happen that some code would be wrong, since it is tested with a newer compiler, just not with an older one.

The advantage is that it's easier to keep the tests running. Right now they are broken.

How do you propose to test older compilers and dependencies if Conda can't do it? Preinstall dependencies as binaries by hand (I think we used to do that on Windows)?

**Comment by SylvainCorlay** [comment]
In any case, this is sorta unrelated to the binder question :)

I am just going to all known binder making use of QuantStack's xeus to make sure that everything works.

I checked that this works on my branch.

**Comment by certik** [comment]
Sort of. Master only fails two tests:

https://travis-ci.org/symengine/symengine/builds/490679972

but your PR breaks three:

https://travis-ci.org/symengine/symengine/builds/488915832

In particular, this test is newly broken by this PR:

https://travis-ci.org/symengine/symengine/jobs/488915856

Any ideas why? I don't want to introduce more failures into master.

File: binder/environment.yml [hunk 1]
```diff
@@ -1,13 +1,11 @@
...
-  - xeus-cling=0.4.7
+  - xeus-cling=0.4.11
+  - llvmdev=5
   - gmp
   - mpfr
   - mpc
   - cmake
   - arb
-  - llvmdev
-  - notebook
```
Comment (isuruf) [inline binder/environment.yml]: @SylvainCorlay, why is this removed?

**Comment by isuruf** [comment]
Tests pass now. LLVM was installed using apt-get and updated the osx image as xcode6.4 image was deprecated.
